### PR TITLE
Object store/ gui

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Updated GUI components to work with Amazon AWS S3 and Microsoft Azure object store.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Review by tom114 — CHANGES_REQUESTED** [review]
Haven't tested its function yet but a couple quick comments and some unused variables.  I'll actually test it soon.

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageUtility.java [hunk 1]
```diff
@@ -113,4 +113,11 @@
 	 */
 	public String getFileExtension(List<? extends SequencingObject> sequencingObjects) throws IOException;
 
+	/**
+	 * Read the bytes for a file
+	 *
+	 * @param file The path to the file
+	 * @return the bytes for the file
+	 */
+	public byte[] readAllBytes(Path file);
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageUtility.java:122]: Yup all that is used for is the FastQC images and any analysis outputs which are images. Looks like it is used in FileUtilities.java as well in the createAnalysisOutputFileZippedResponse method which was changed from `byte[] bytes = objectMapper.writeValueAsBytes(file);` to `byte[] bytes = file.getBytesForFile();` to work with out object store stuff. Anywhere the files are larger they are streamed using the getFileInputStream method.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/files/SequenceFileController.java [hunk 2]
```diff
@@ -88,7 +87,7 @@ public void downloadSequenceFile(@PathVariable Long sequencingObjectId, @PathVar
 		SequenceFile sequenceFile = sequencingObject.getFileWithId(sequenceFileId);
 		Path path = sequenceFile.getFile();
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/files/SequenceFileController.java]: Removed in 812a1b86c4c0abf1994668d24b8080b497cf3d34

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisAjaxController.java [hunk 3]
```diff
@@ -1093,8 +1094,11 @@ public AnalysisTreeResponse getNewickTree(@PathVariable Long submissionId, Local
 					}
 				}
 			} catch (NoSuchFileException e) {
-				logger.debug("File was not found: " + e.toString());
+				logger.error("File was not found: " + e.toString());
+			} catch (IOException e) {
+				logger.error("Unable to read input stream for file");
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisAjaxController.java:1099]: Add the throwable cause here in the error logging so we can see the stack trace please.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/files/SequenceFileController.java [hunk 4]
```diff
@@ -86,9 +85,14 @@ public void downloadSequenceFile(@PathVariable Long sequencingObjectId, @PathVar
...
 		SequencingObject sequencingObject = sequencingObjectService.read(sequencingObjectId);
 		SequenceFile sequenceFile = sequencingObject.getFileWithId(sequenceFileId);
-		Path path = sequenceFile.getFile();
 		response.setHeader("Content-Disposition", "attachment; filename=\"" + sequenceFile.getLabel() + "\"");
-		Files.copy(path, response.getOutputStream());
+
+		try(InputStream inputStream = sequenceFile.getFileInputStream()) {
+			inputStream.transferTo(response.getOutputStream());
+		} catch (IOException e) {
+			logger.error("Unable to read input stream from file", e);
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/files/SequenceFileController.java:93]: Same here.  Add the cause.

**Review by tom114 — CHANGES_REQUESTED** [review]

**Review by tom114 — APPROVED** [review]

**Pull request merged**